subsys/net/lib/dhcpv4/dhcpv4_server.c

```diff
@@ -667,13 +683,163 @@ static uint32_t dhcpv4_get_lease_time(uint8_t *options, uint8_t optlen)
 	return CONFIG_NET_DHCPV4_SERVER_ADDR_LEASE_TIME;
 }
 
+#if defined(DHCPV4_SERVER_ICMP_PROBE)
+static void dhcpv4_probe_address(struct dhcpv4_server_ctx *ctx,
+				 struct dhcpv4_addr_slot *slot)
+{
+	struct sockaddr_in dest_addr = {
+		.sin_family = AF_INET,
+		.sin_addr = slot->addr,
+	};
+	int ret;
+
+	ret = net_icmp_send_echo_request(&ctx->probe_ctx.icmp_ctx, ctx->iface,
+					 (struct sockaddr *)&dest_addr,
+					 NULL, ctx);
+	if (ret < 0) {
+		LOG_ERR("Failed to send ICMP probe");
+	}
```

Comment:
Do we still need to wait for a timeout if this already fails?

Reply:
True, no point waiting there. I guess it'd make sense to stop the probing at that point, and wait for the next Discovery?